tests/test_visual.py

```diff
@@ -101,21 +105,35 @@ def test_plot_bboxes():
     img = Image.open(
         str(cur_dir / "mock_data" / "simrun" / "captures" / "camera_000.png")
     )
+    label_mappings = {1: "car", 2: "tree", 3: "light"}
     boxes = [
-        BBox2D(label="car", x=1, y=1, w=2, h=3),
-        BBox2D(label="tree", x=7, y=6, w=3, h=4),
-        BBox2D(label="light", x=2, y=6, w=2, h=4),
-    ]
-    colors = [
-        ImageColor.getcolor("green", "RGB"),
-        ImageColor.getcolor("red", "RGB"),
-        ImageColor.getcolor("green", "RGB"),
+        BBox2D(label=1, x=1, y=1, w=2, h=3),
+        BBox2D(label=1, x=7, y=6, w=3, h=4),
+        BBox2D(label=1, x=2, y=6, w=2, h=4),
     ]
+    colors = ["green", "red", "green"]
 
     with patch(
-        "datasetinsights.stats.visualization.plots.ImageDraw.Draw"
+        "datasetinsights.stats.visualization.plots.add_single_bbox_on_image"
     ) as mock:
-        instance = mock.return_value
-        plot_bboxes(img, boxes, colors)
-        assert instance.rectangle.call_count == len(boxes)
-        assert instance.multiline_text.call_count == len(boxes)
+        plot_bboxes(img, boxes, label_mappings=label_mappings, colors=colors)
+        assert mock.call_count == len(boxes)
+
+
+@patch("datasetinsights.evaluation_metrics.confusion_matrix.Records")
+def test_match_boxes(mock_record):
+    match_results = [(0.5, True), (0.6, False), (0.7, True)]
+    expected_colors = ["green", "red", "green"]
+    mock_record.return_value.match_results.return_value = match_results
+    colors = match_boxes(None, None)
+    for i in range(len(colors)):
+        assert colors[i] == expected_colors[i]
+
+
+@patch(
+    "datasetinsights.stats.visualization.bbox2d_plot._add_single_bbox_on_image"
```

Comment:
I agree that the method itself should not be tested as it is a private method, @adason might disagree. But the behavior that code performs should be tested. Since we are copying the code from another repo mean that we are going to maintain the code. How will we validate that if code changes happens that the behavior is as expected. Furthermore I don't see any tests in the public repo, so I still think we should have some.

Reply:
A test should test the behavior of the code and not the code implementation itself.

Reply:
I will add test the following behavior: make sure that `cv2.rectangle` is called once if `label=None`, and `cv2.rectangle` is called twice if we set the `label`.